added ProjectChampions relation that connects User and Project types

mugwump303 opened the pull request:
This PR closes #162 

Adds ProjectChampion relation that connects two types (User and Project).

mugwump303 committed:
added ProjectChampions relation that connects User and Project types

wdoug commented on server/types.graphql:
```diff
@@ -17,6 +17,7 @@ type User @model {
   flowdockName: String
   description: String
   role: UserRole @defaultValue(value: REGULAR)
+  projects: Project @relation(name: "ProjectChampions")
```
We want a many-to-many relationship here since members can have multiple projects they are championing/have championed. I think we might also want to have this called `projectsChampioned` or something more specific since we could have other relationships to projects from users as well.

mugwump303 committed:
Changed projects field in User type to projectsChampioned. Turned return type of that field to a list of Project types.

wdoug commented:
I just went to test this, and it looks like we need to explicity add the permission to do this so we can call `addToProjectChampions`

mugwump303 committed:
added permissions to yml file. added permission queries to permissions folder.
anybody can create a project and add users to project champions. champions can update projects and remove themselves from project. admins can remove anybody from project

mugwump303 committed:
added permission to allow admins to update projects without being champions

wdoug reviewed (changes requested):
I get this error when trying to edit a project as an admin:

<img width="518" alt="screen shot 2018-04-19 at 8 44 27 pm" src="https://user-images.githubusercontent.com/5432102/39028177-921e6f04-4412-11e8-99fa-b9a7e7ccb317.png">

wdoug commented:
I wonder if we want to temporarily keep allowing all users to update projects. That makes it so we can merge this pull request without having to make any changes to the front-end

mugwump303 committed:
fixed champion permissions. allow all authenticated the ability to update projects

Commit:
Update schema

Commit:
Allow users to edit any project field for now
